### PR TITLE
Add Course::ExpRecord model

### PR DESCRIPTION
**Pull request by kxmbrian**

### CODE REVIEW

File: app/models/course/exp_record.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  actable
+  stampable
+
+  belongs_to :course_user, inverse_of: :exp_records
+
+  validates :course_user, presence: true
+  validates :exp_awarded, numericality: { only_integer: true }
+  validates :reason, presence: true, if: :is_manual_exp?
+
+  def is_manual_exp?
```
Comment (allenwq) [inline app/models/course/exp_record.rb]: Don't need <code>is</code>, <code>manual_exp?</code> looks more ruby

File: spec/models/course/exp_record_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+RSpec.describe Course::ExpRecord, type: :model do
+  it { is_expected.to belong_to(:course_user).inverse_of(:exp_records) }
+  it { is_expected.to validate_presence_of(:course_user) }
+  it { is_expected.to validate_numericality_of(:exp_awarded).only_integer }
+
+  let!(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    context 'when manual' do
+      let(:manual_exp_record) { FactoryGirl.build :manual_exp_record }
```
Comment (allenwq) [inline spec/models/course/exp_record_spec.rb]: Use a subject here

File: spec/models/course/exp_record_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+      it { expect(manual_exp_record).to be_valid }
```
Comment (allenwq) [inline spec/models/course/exp_record_spec.rb]: then can use
```
it { is_expected.to be_valid }
```

File: spec/models/course/exp_record_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+      it '.is_manual_exp? is true' do
+        expect(manual_exp_record.is_manual_exp?).to be_truthy
```
Comment (allenwq) [inline spec/models/course/exp_record_spec.rb]: same here, after you renamed the function:

```
it { is_expected.to be_manual_exp }
```

File: spec/models/course/exp_record_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      end
+
+      context 'has no reason' do
```
Comment (allenwq) [inline spec/models/course/exp_record_spec.rb]: By convention context should be followed by <code>with</code> or <code>when</code>
so
```
context 'when where is no reason'
```

File: app/models/course_user.rb [hunk 6]
```diff
@@ -1,5 +1,8 @@
 class CourseUser < ActiveRecord::Base
   belongs_to :user, inverse_of: :course_users
   belongs_to :course, inverse_of: :course_users
+  has_many :exp_records, class_name: 'Course::ExpRecord',
```
Comment (allenwq) [inline app/models/course_user.rb]: <code> Course::ExpRecord.name </code>, Thus it will raise an exception when your class does not exist.

**Comment by kxmbrian** [comment]
PR is replaced by #183.

**Pull request closed without merging**